core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/keystore/persistence/KeyStoreRegistryPersistenceManager.java

```diff
@@ -0,0 +1,421 @@
+    }
+
+    /**
+     * Get the key store from the registry.
+     *
+     * @param keyStoreName Name of the key store.
+     * @param tenantId     Tenant Id.
+     * @return Key store.
+     * @throws SecurityException If an error occurs while getting the key store.
+     */
+    public KeyStore getKeyStore(String keyStoreName, int tenantId) throws SecurityException {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        if (!isKeyStoreExistsInRegistry(keyStoreName, registry)) {
+            throw new SecurityException("Key Store with a name: " + keyStoreName + " does not exist.");
+        }
+
+        try {
+            String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+            Resource resource = registry.get(path);
+            byte[] bytes = (byte[]) resource.getContent();
+            KeyStore keyStore = KeystoreUtils.getKeystoreInstance(resource
+                    .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
+            char[] passwordChar = getKeyStorePasswordFromRegistryResource(resource, keyStoreName);
+            ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
+            keyStore.load(stream, passwordChar);
+            resource.discard();
+            return keyStore;
+        } catch (RegistryException | KeyStoreException | NoSuchProviderException | IOException | CertificateException |
+                 NoSuchAlgorithmException e) {
+            throw new SecurityException("Error getting key store: " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Method to retrieve list of keystore metadata of all the keystores in a tenant.
+     *
+     * @param tenantId Tenant Id.
+     * @return List of KeyStoreMetaData objects.
+     * @throws SecurityException If an error occurs while retrieving the keystore data.
+     */
+    public List<KeyStoreMetadata> listKeyStores(int tenantId) throws SecurityException {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        List<KeyStoreMetadata> metadataList = new ArrayList<>();
+        try {
+            if (!registry.resourceExists(RegistryResources.SecurityManagement.KEY_STORES)) {
+                return metadataList;
+            }
+            Collection keyStoreCollection = (Collection) registry.get(RegistryResources.SecurityManagement.KEY_STORES);
+            String[] keyStoreFullNames = keyStoreCollection.getChildren();
+
+            for (String keyStoreFullName : keyStoreFullNames) {
+                if (RegistryResources.SecurityManagement.PRIMARY_KEYSTORE_PHANTOM_RESOURCE.equals(keyStoreFullName)) {
+                    continue;
+                }
+                metadataList.add(getKeyStoreMetadata(keyStoreFullName, registry, tenantId));
+            }
+            return metadataList;
+        } catch (RegistryException e) {
+            String msg = "Error when getting keyStore metadata.";
+            throw new SecurityException(msg, e);
```

Comment:
Lets directly use the string literal in the exception, rather using a variable?

Reply:
Addressed with https://github.com/wso2/carbon-kernel/compare/6c4a7e5fc5efca69a114e5d3a08289b8dc809be7..3a9a8a70b486cbcffcd4d4da8c10588bd93f0358